### PR TITLE
Add s3 backups option. Add role dependencies.

### PR DESCRIPTION
**Pull request by Nonsense331**

### CODE REVIEW

File: ansible/roles/mtpereira.passenger/tests/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# -*- mode: ruby -*-
```
Comment (samsonasu) [inline ansible/roles/mtpereira.passenger/tests/Vagrantfile]: Should we remove this stuff from here?  does it matter?

File: ansible/roles/zenoamaro.postgresql/README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+It can additionally be used as a playbook for quickly provisioning hosts.
+
+Vagrant machines are provided to produce a boxed install of PostgreSQL or a VM for integration testing.
+
+
+Supports
+--------
+Supported PostgreSQL versions:
+
+- PostgreSQL 9.4
```
Comment (samsonasu) [inline ansible/roles/zenoamaro.postgresql/README.md]: is there an updated version of this that supports 9.5/6 and ubuntu 16.04?

File: lib/subspace/commands/provision.rb [hunk 3]
```diff
@@ -12,7 +12,10 @@ def run
   private
 
   def update_ansible_cfg
-    template "ansible.cfg"
```
Comment (samsonasu) [inline lib/subspace/commands/provision.rb]: why not just rerun the template?

File: template/provision/ansible.cfg.erb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 [defaults]
 inventory = hosts
 forks     = 10
-roles_path = <%= File.join(gem_path, 'ansible', 'roles') %>:/etc/ansible/roles
+roles_path = <%= File.join(gem_path, 'ansible', 'roles') %>
```
Comment (samsonasu) [inline template/provision/ansible.cfg.erb]: I think we should maybe also still include /etc/ansible/roles? does that mess with anything if they are in there twice?

File: ansible/roles/postgresql/templates/backup.sh [hunk 5]
```diff
@@ -36,15 +38,28 @@ else
...
-# then clean old ones
-# hourly - for 48 hours
-find $storage/backup.hourly/ -mmin +2880 -exec rm -rv {} \;
-# daily - keep for 14 days
-find $storage/backup.daily/ -mtime +14 -exec rm -rv {} \;
-# weekly - keep for 60 days
-find $storage/backup.weekly/ -mtime +60 -exec rm -rv {} \;
-# monthly - keep for 300 days
-find $storage/backup.monthly/ -mtime +300 -exec rm -rv {} \;
+if [ "$bucket" != "disabled" ] ; then
```
Comment (samsonasu) [inline ansible/roles/postgresql/templates/backup.sh]: did you test this?

**Comment by samsonasu** [comment]
🚢

File: ansible/roles/postgresql/templates/backup.sh (hunk 5 above)
Comment (Nonsense331) [inline ansible/roles/postgresql/templates/backup.sh]: yes

File: ansible/roles/zenoamaro.postgresql/README.md (hunk 2 above)
Comment (Nonsense331) [inline ansible/roles/zenoamaro.postgresql/README.md]: no

**Pull request merged**